src/user/profile/ReadMe.js

```diff
@@ -0,0 +1,245 @@
+import React, { Component } from "react";
+import { Card, Button } from "react-bootstrap";
+import "./ReadMe.scss";
+import { Editor } from "@tinymce/tinymce-react";
+import { updateProfile, getProfile } from "../../actions/usersAction";
+import { connect } from "react-redux";
+
+class ReadMe extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      longDescription: true,
+      displayEditor: false,
+      currentText: "",
+      firstName: "",
+      lastName: "",
+      shortDesc: "",
+      location: "",
+      github: "",
+      fb: "",
+      linkedIn: "",
+      designation: "",
+      website: "",
+      longDesc: "",
+      canChangeName: "",
+    };
+  }
+
+  componentDidMount() {
+    const  userProfile  = this.props.userProfile;
+    const about = userProfile.info?.about;
+    let longDescription = true;
+    if (!about.hasOwnProperty("longDescription")) {
+      longDescription = false;
+    }
+
+    this.setState({
+      firstName: userProfile.name?.firstName,
+      lastName: userProfile.name?.lastName,
+      website: about?.website,
+      designation: about?.designation,
+      longDesc: about?.longDescription,
+      shortDesc: about?.shortDescription,
+      location: about?.location,
+      longDescription: longDescription,
+    });
+  }
+
+  handleCreateReadMe = () => {
+    this.setState({
+      displayEditor: true,
+    });
+  };
+
+  handleEditorChange = (content, editor) => {
+    this.setState({
+      currentText: content,
+    });
+  };
+
+  handleSave = () => {
+    const {
+      firstName,
+      lastName,
+      designation,
+      location,
+      website,
+      shortDesc,
+      currentText,
+    } = this.state;
+    const info = {
+      name: {
+        firstName,
+        lastName,
+      },
+      info: {
+        about: {
+          shortDescription: shortDesc,
+          longDescription: currentText,
+          designation,
+          location,
+          website,
+        },
+      },
+    };
+
+    this.props.updateProfile(info);
+    this.setState(
+      {
+        displayEditor: false,
+      },
+      () => {
+        this.setState({ longDescription: true });
+      }
+    );
+  };
+
+  handleEdit = () => {
+    this.setState(
+      {
+        longDescription: false,
+      },
+      () => {
+        this.setState({
+          displayEditor: true,
```

Comment:
same comment. Also maybe we can have only one function as we just setting boolean values to opposite of what they are or we could accept arguments to perform the desired changes than having two different functions handleSave and handleEdit.

Reply:
@ksraj123 I don't think we can extract this to a separate component since the order of the two setStates are different.

Reply:
Hmm.. If we were setting state in one go then this would have worked. Anyways, its cool if we absolutely need to do this one after another.